sympy/matrices/tests/test_eigen.py

```diff
@@ -214,6 +214,18 @@ def test_eigenvals():
     # CRootOf may not be unique.
     assert m.eigenvals()
 
+    A = Matrix([
+        [1, 1, 0, 1, -1, 0, 1],
+        [0, 2, 1, 0, 1, -1, 0],
+        [1, 0, 1, 1, 0, 1, -1],
+        [1, -1, 0, 0, 1, 0, 1],
+        [0, 1, 1, -1, 1, 1, 0],
+        [1, 0, -1, 1, 0, 1, 1],
+        [0, 1, 0, 1, -1, 0, 2]
+    ])
+    eigs = A.eigenvals()
+    assert sum(eigs.values()) == A.charpoly().degree()
```

Comment:
It is not obvious to me why this equality should hold.

I think it would be better to compare with `A.trace()` rather than `A.charpoly().degree()`. For any square matrix the sum of the eigenvalues (counted with multiplicity) should be equal to the trace. That would seem more obviously correct to me.